### ISSUE TITLE
Decrypted strings are `b'` format

### ISSUE THREAD

**Issue by sitya** [OpenConext/Stepup-Deploy#76]
All decrypted strings are given in byte format, eg. `b'-----BEGIN PRIVATE KEY-----\nMIIEvQIBADANBgkqh\nfdsfsdfsfsd...` or passwords also. I am using Python 2.7.14 (with ansible 2.7) while python 3 is not allowed to use for stepup-deploy where strings with this format are autoconverted. Do you have any idea how to decode them? Is there any jinja2 filter to do this? Or in `custom_plugins.py` could do it? Thanks

**Comment by pmeulen** [comment]
What do you mean by "in byte format"?

The `vault` filter that is in `custom_plugins.py` should give you back the decoded string without adding any `b`, '`' or `\n`. You can use them directly in jinja2 templates.

Could it be that something went wrong during the encoding process, and that the extra characters where added then? Something like that could easily happen if `/usr/bin/python` refers to python3 on your system. The encrypt.sh and encrypt_file.sh scripts are python and have a shebang that refers to the system `python` (e.g. /usr/bin/python). `create_new_environment.sh` and the `gen_*.sh` calls these scripts. A virtualenv won't save you, because it is the location of the python interpreter that determines the python environment that is used.

**Comment by sitya** [comment]
Thanks for the quick reply!

I have double checked now:

```
$ /usr/bin/env python
Python 2.7.14 (default, Mar 14 2018, 13:16:32)
...
>>>
```

I tried to decrypt manually with: 
```
$ scripts/encrypt-file.sh -d --file ../stepup-test-example-org/ssl_cert/gateway.key ../stepup-test-example-org/stepup-ansible-keystore
``` 
and the result is fine, good PEM format private key, so I assume `create_new_environment.sh` did well. I made a debug task in ansible, and when it dumps the value, it is already in `b'....` format. I have no ide where it could go wrong.

**Comment by sitya** [comment]
I am not sure, if it is a workaround, or it was a bug, or some misconfiguration in my desktop, but the following change solved the problem:

```diff
diff --git a/filter_plugins/custom_plugins.py b/filter_plugins/custom_plugins.py
index 4094962..6ca0c44 100644
--- a/filter_plugins/custom_plugins.py
+++ b/filter_plugins/custom_plugins.py
@@ -19,7 +19,7 @@ crypter = keyczar.Crypter.Read(expanded_keydir)
 sys.stdout.write(crypter.Decrypt("%s"))
   """ % (keydir, encrypted)
   from subprocess import check_output
-  return check_output(["python", "-c", method])
+  return check_output(["python", "-c", method], universal_newlines=True)
```

**Comment by pmeulen** [comment]
Just a hunch. Is your ansible using python 2 or 3? (What does `ansible --version` say)

**Comment by sitya** [comment]
Good catch!

```
$ ansible --version
ansible 2.7.0
  config file = ~/.ansible.cfg
  configured module search path = ['~/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.0 (default, Jul  5 2018, 22:07:50) [Clang 9.1.0 (clang-902.0.39.2)]
```

I will try to make ansible to use 2.7.14 then we will see. Thanks.

**Comment by pmeulen** [comment]
OK.

I'll gladly accept fixes for python 3, but these will need to work for python 2.7 as well because it will be a while before everyone / everything is using python 3. Supporting both should be possible, you can check the python version for instance.

In stepup all secrets are text based, so your fix should work fine. I'm not sure if it will work correctly when storing binary blobs (e.g. DER encoded keys).

I'm closing this issue, feel free to reopen if you have more to say.

**Comment by pmeulen** [comment]
Both python 2 and 3 should work equally well now. Fixed by #77
